lib/rubocop/cop/mixin/uncommunicative_name.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality shared by Uncommunicative cops
+    module UncommunicativeName
+      CASE_MSG = 'Only use lowercase characters for %<name_type>s.'.freeze
+      NUM_MSG = 'Do not end %<name_type>s with a number.'.freeze
+      LENGTH_MSG = '%<name_type>s must be longer than %<min>s ' \
+                   'characters.'.freeze
+
+      def check(node, args)
+        args.each do |arg|
+          source = arg.source
+          case_offense(node, arg) if uppercase?(source)
+          num_offense(node, arg) if ends_with_num?(source)
+          length_offense(node, arg) unless long_enough?(source)
+        end
+      end
+
+      def case_offense(node, arg)
+        add_offense(node, location: arg.source_range,
+                          message: format(CASE_MSG, name_type: name_type(node)))
+      end
+
+      def uppercase?(name)
+        name =~ /[A-Z]/
+      end
+
+      def name_type(node)
+        @name_type ||= begin
+          case node.type
+          when :block then 'block parameter'
+          when :def then 'method argument'
+          end
+        end
+      end
+
+      def num_offense(node, arg)
+        add_offense(node, location: arg.source_range,
+                          message: format(NUM_MSG, name_type: name_type(node)))
+      end
+
+      def ends_with_num?(name)
+        name[-1] =~ /\d/
+      end
+
+      def length_offense(node, arg)
+        add_offense(node, location: arg.source_range,
+                          message: format(LENGTH_MSG,
+                                          name_type: name_type(node).capitalize,
+                                          min: min_length))
+      end
+
+      def long_enough?(name)
+        name.size >= min_length
+      end
+
+      def min_length
+        cop_config['MinParamNameLength'] ||
+          cop_config['MinArgNameLength']
```

Comment:
This method is confusing. We will be able to set `MinParamNameLength` to `Naming/UncommunicativeMethodArgName:`. I think it is a bug. `Naming/UncommunicativeMethodArgName` should only use `MinArgNameLength`, do not use `MinParamNameLength`. 

For example:


```ruby
# in the mixin
def min_length
  cop_config[MIN_LENGTH_KEY]
end

# in the cops
class UncommunicativeMethodArgName
  MIN_LENGTH_KEY = 'MinArgNameLength'
end
```

Reply:
Great point! Thanks!

Reply:
I wasn't able to do exactly this. Since the mixin is a `Module`, it didn't have access to constants from the Class it extends.